src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -321,19 +338,24 @@ public ResponseEntity<Exercise> getExerciseDetails(@PathVariable Long exerciseId
             quizExercise.setQuizBatches(null);
             quizExercise.setQuizBatches(quizBatchService.getQuizBatchForStudentByLogin(quizExercise, user.getLogin()).stream().collect(Collectors.toSet()));
         }
-        if (exercise instanceof ProgrammingExercise programmingExercise) {
-            // TODO: instead fetch the policy without programming exercise, should be faster
-            SubmissionPolicy policy = programmingExerciseRepository.findByIdWithSubmissionPolicyElseThrow(programmingExercise.getId()).getSubmissionPolicy();
-            programmingExercise.setSubmissionPolicy(policy);
-        }
         // TODO: we should also check that the submissions do not contain sensitive data
 
         // remove sensitive information for students
         if (!isAtLeastTAForExercise) {
             exercise.filterSensitiveInformation();
         }
 
-        return ResponseEntity.ok(exercise);
+        IrisCombinedSettingsDTO irisSettings = irisSettingsService.map(service -> service.getCombinedIrisSettingsFor(exercise, service.showMinimalSettings(exercise, user)))
+                .orElse(null);
+        PlagiarismCaseInfoDTO plagiarismCaseInfo = plagiarismCaseService.getPlagiarismCaseInfoForExerciseAndUser(exercise.getId(), user.getId()).orElse(null);
+
+        if (exercise instanceof ProgrammingExercise programmingExercise) {
+            Set<ExerciseHint> activatedExerciseHints = exerciseHintService.getActivatedExerciseHints(programmingExercise, user);
+            Set<ExerciseHint> availableExerciseHints = exerciseHintService.getAvailableExerciseHints(programmingExercise, user);
```

Comment:
This method uses a lot of database calls and as far as I can tell, we don't check first if there are any hints available. So we should probably wrap this into an if-statement that first checks if we actually need these database calls.
One option would be to load the exercise hints in the method first and check if there are any